components/global/core/io.cellery.observability.k8s.client/src/main/java/io/cellery/observability/k8s/client/ComponentsEventSource.java

```diff
@@ -0,0 +1,259 @@
+import io.cellery.observability.k8s.client.cells.CellList;
+import io.cellery.observability.k8s.client.cells.DoneableCell;
+import io.cellery.observability.k8s.client.cells.model.HTTP;
+import io.cellery.observability.k8s.client.cells.model.IngressTypes;
+import io.cellery.observability.k8s.client.cells.model.TCP;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.CustomResourceDefinitionList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.exception.ConnectionUnavailableException;
+import org.wso2.siddhi.core.stream.input.source.Source;
+import org.wso2.siddhi.core.stream.input.source.SourceEventListener;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * This class implements the Event Source which can be used to listen for K8s cell changes.
+ */
+@Extension(
+        name = "k8s-cellery-components",
+        namespace = "source",
+        description = "This is an event source which emits events upon changes to Cellery cells deployed as" +
+                "Kubernetes resource",
+        examples = {
+                @Example(
+                        syntax = "@source(type='k8s-cellery-components', @map(type='keyvalue', " +
+                                "fail.on.missing.attribute='false'))\n" +
+                                "define stream K8sComponentEventSourceStream (cell string, component string," +
+                                " creationTimestamp long, lastKnownActiveTimestamp long, Ingress_Types string" +
+                                ", action string)",
+
+                        description = "This will listen for kubernetes cell events and emit events upon changes " +
+                                "to the pods"
+                )
+        }
+)
+public class ComponentsEventSource extends Source {
+
+    private static final Logger logger = Logger.getLogger(ComponentsEventSource.class.getName());
+
+    private static final String CELL_CRD_GROUP = "mesh.cellery.io";
+    private static final String CELL_CRD_NAME = "cells." + CELL_CRD_GROUP;
+    private static final String INGRESS_TYPES = "Ingress_Types";
```

Comment:
As mentioned above, let's move all the attribute related constants to the constants class.